app/uk/gov/hmrc/pillar2externalteststub/models/uktr/UKTRSubmissionData.scala

```diff
@@ -34,15 +34,21 @@ case class UKTRSubmissionData(
 
 object UKTRSubmissionData {
   implicit val uktrSubmissionDataFormat: OFormat[UKTRSubmissionData] = Json.format[UKTRSubmissionData]
-  val amountErrorMessage =
-    " must be Numeric, positive, with at most 2 decimal places, and less than or equal to 13 characters, including the decimal place."
-
-  private def isValidUKTRAmount(number: String): Boolean = {
-    val pattern = """^\d{1,13}\.{0,1}\d{0,2}$""".r
-    number match {
-      case pattern() if number.length <= 13 && number.toDouble > 0 => true
-      case _                                                       => false
-    }
+
+  private val boundaryUKTRAmount                    = BigDecimal("9999999999999.99")
+  private def isValidUKTRAmount(number: BigDecimal) = number >= 0 && number <= boundaryUKTRAmount && number.scale <= 2
+  private val amountErrorMessage                    = s"must be a number between 0 and $boundaryUKTRAmount with up to 2 decimal places"
+
+  private def obligationMTTRule(plrReference: String): ValidationRule[UKTRSubmissionData] = ValidationRule { data =>
```

Comment:
it would be good to combine this with the obligationMTTRule for NilReturn too.

Reply:
There's negligible difference between nil and liability returns validations. We should combine the rules into UKTRSubmission once this epic is over and we have a complete coverage and understanding of business rules.